src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/AzureAttestationBasedEnclaveProvider.cs

```diff
@@ -213,7 +213,7 @@ public AzureAttestationInfo(byte[] attestationInfo)
                 }
                 catch (Exception exception)
                 {
-                    throw new AlwaysEncryptedAttestationException(string.Format(Strings.FailToParseAttestationInfo, exception.Message));
+                    throw SQL.AttestationFailed(string.Format(Strings.FailToParseAttestationInfo, exception.Message));
```

Comment:
exceptions can be passed as inner exception to SqlException? not just the message. same on line 429 and 504

Reply:
It is possible that passing the full exception would expose an internal exception type to users. 

If the exception that can be caught here might come from users code then it seems like a good idea to pass it back to them as the inner exception because it might provide extra context information about the failure. If the exception can only be internal to this library then I think simply concatenating the message is sufficient.